drv/transceivers-api/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! API crate for QSFP transceiver managment
+
+#![no_std]
+
+use derive_idol_err::IdolError;
+use drv_fpga_api::FpgaError;
+use userlib::{sys_send, FromPrimitive};
+
+#[derive(Copy, Clone, Debug, FromPrimitive, Eq, PartialEq, IdolError)]
+pub enum TransceiversError {
+    FpgaError = 1,
+    InvalidPortNumber = 2,
+    InvalidNumberOfBytes = 3,
+}
+
+impl From<FpgaError> for TransceiversError {
+    fn from(_: FpgaError) -> Self {
+        Self::FpgaError
+    }
+}
+
+#[derive(Copy, Clone, zerocopy::FromBytes, zerocopy::AsBytes)]
+#[repr(C)]
+pub struct ModulesStatus {
+    pub enable: u32,
+    pub reset: u32,
+    pub lpmode_txdis: u32,
+    pub power_good: u32,
+    pub power_good_timeout: u32,
+    pub present: u32,
+    pub irq_rxlos: u32,
+}
+
+impl From<[u32; 7]> for ModulesStatus {
```

Comment:
Because you've derived `FromBytes`, you could simplify this implementation to

```rust
ModulesStatus::read_from(data.as_bytes()).unwrap()
```

However, looking at the bigger picture, I'd consider
- removing the `impl From<...>` entirely
- make `Transceivers::get_modules_status` return a `ModuleStatus` instead of `[u32; 7]`
    - Do the `ModuleStatus::read_from(data.as_bytes()).unwrap()` in that function

(so we're _never_ passing around `[u32; 7]`, which is an unclear data type)

Reply:
Thanks for both of the recommended paths here. As you mention, given the bigger picture here, I will address this with your second recommendation.